src/OpenTelemetry/Trace/SamplingResult.cs

```diff
@@ -117,11 +117,23 @@ public override bool Equals(object? obj)
         /// <inheritdoc/>
         public override int GetHashCode()
         {
-            var result = 1;
-            result = (31 * result) + this.Decision.GetHashCode();
-            result = (31 * result) + this.Attributes.GetHashCode();
-            result = this.TraceStateString == null ? result : (result * 31) + this.TraceStateString.GetHashCode();
-            return result;
+#if NETSTANDARD2_1 || NET6_0_OR_GREATER
+            HashCode hashCode = default;
+            hashCode.Add(this.Decision.GetHashCode());
+            hashCode.Add(this.Attributes.GetHashCode());
+            hashCode.Add(this.TraceStateString?.GetHashCode() ?? 0);
+
+            var hash = hashCode.ToHashCode();
+#else
+            var hash = 17;
+            unchecked
+            {
+                hash = (31 * hash) + this.Decision.GetHashCode();
+                hash = (31 * hash) + this.Attributes.GetHashCode();
+                hash = this.TraceStateString == null ? hash : (hash * 31) + this.TraceStateString.GetHashCode();
```

Comment:
I don't think it is a big deal, but if you want to be consistent with the others...

```csharp
hash = (31 * hash) + this.TraceStateString?.GetHashCode() ?? 0;
```

Reply:
Update this.

I have also made this change for `MetricStreamIdentity` class as well:

```csharp
hash = (hash * 31) + this.Unit?.GetHashCode() ?? 0;
hash = (hash * 31) + this.Description?.GetHashCode() ?? 0;
hash = (hash * 31) + this.ViewId ?? 0;
hash = (hash * 31) + (this.TagKeys != null ? StringArrayComparer.GetHashCode(this.TagKeys) : 0);
```